### PR TITLE
Add `commands` option

### LINKED ISSUE #69: Support subcommands

### PR DESCRIPTION
**Pull request by skoshx**
This commit adds support for subcommands.

Fixes #69

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -68,6 +70,30 @@ export interface Options<Flags extends AnyFlags> {
 	*/
 	readonly flags?: Flags;
 
+	/**
+  Define subcommands.
```
Comment (sindresorhus) [inline index.d.ts:74]: Use tab-indentation.

File: index.d.ts [hunk 2]
```diff
@@ -68,6 +70,30 @@ export interface Options<Flags extends AnyFlags> {
...
+  The key is the name of the subcommand and the value is a function that returns an instance of `meow`.
+
+  The following values get passed to the subcommand function:
+  - `helpText`: The help text of parent `meow` instance.
+  - `options`: The options from the parent `meow` instance.
+
+  @example
+  ```
+  commands: {
+    unicorn: (helpText, options) => meow({
```
Comment (sindresorhus) [inline index.d.ts:85]: The `helpText` argument is moot. `helpText` can already be passed in `options`.

File: index.d.ts (hunk 1 above)
Comment (sindresorhus) [reply index.d.ts:74]: The description needs to be more extensive. It should describe how it works (that it doesn't actually call any commands for you, etc).

File: index.d.ts [hunk 3]
```diff
@@ -68,6 +70,30 @@ export interface Options<Flags extends AnyFlags> {
...
+
+  The key is the name of the subcommand and the value is a function that returns an instance of `meow`.
+
+  The following values get passed to the subcommand function:
+  - `helpText`: The help text of parent `meow` instance.
+  - `options`: The options from the parent `meow` instance.
+
+  @example
+  ```
+  commands: {
```
Comment (sindresorhus) [inline index.d.ts:84]: The example should be complete and runnable. I think it's better to define the subcommand function at the top-level, for readability.

File: index.js [hunk 4]
```diff
@@ -205,6 +205,16 @@ const meow = (helpText, options = {}) => {
...
 	}
 
+	// Subcommands
+	const commands = {};
+	for (const [command, meowInstance] of Object.entries(options.commands ?? {})) {
+		if (input[0] !== command) {
+			continue;
+		}
+
+		commands[command] = meowInstance(helpText, {...options, argv: process.argv.slice(3), commands: {}});
```
Comment (sindresorhus) [inline index.js:215]: Pass `helpText` in `options`.

File: test/fixtures/fixture-commands.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import meow from '../../index.js';
+
+const subcommand = (helpText, options = {}) => meow({
+	...options,
+	description: 'Subcommand description',
+	help: `
+    Unicorn command
+    Usage:
+      foo unicorn <input>
+  `,
```
Comment (sindresorhus) [inline test/fixtures/fixture-commands.js:11]: Tab-indentation.

File: test/fixtures/fixture-commands.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env node
+import meow from '../../index.js';
+
+const subcommand = (helpText, options = {}) => meow({
```
Comment (sindresorhus) [inline test/fixtures/fixture-commands.js:4]:
```suggestion
const subcommand = (helpText, options) => meow({
```

The `options` should be guaranteed. Make sure the code guarantees it to exist.

File: index.d.ts (hunk 3 above)
Comment (sindresorhus) [reply index.d.ts:84]: The example should also illustrate how one might check for and call code for the subcommand.

File: index.d.ts [hunk 7]
```diff
@@ -68,6 +70,30 @@ export interface Options<Flags extends AnyFlags> {
...
+    unicorn: (helpText, options) => meow({
+      ...options,
+      description: 'Subcommand description',
+      flags: {
+        unicorn: {alias: 'u'},
+      }
+    })
+  }
+  ```
+  */
```
Comment (sindresorhus) [inline index.d.ts:94]: This also needs to be documented in the readme.

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by backtolife2021** [comment]
@skoshx  This is a very useful feature, this thread seems to have stalled for a long time, is there any progress now?

**Comment by skoshx** [comment]
@backtolife2021 I made the changes requested by Sindre, other than that everything is good to go, just waiting for Sindre's review.

friendly bump :) @sindresorhus

**Comment by skoshx** [comment]
Closing in favor of PR #211

**Pull request closed without merging**